### PR TITLE
Add the functional interface context initializer for square response handling

### LINKED ISSUE #17: Handle the Square customized response
Handle the Square customized response

### PR DESCRIPTION
**Pull request by hamzashoukat94**
Add the context initializer for square customized response handling.

closes #17

### CODE REVIEW

File: src/main/java/io/apimatic/coreinterfaces/type/functional/ContextInitializer.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@FunctionalInterface
+public interface ContextInitializer<ResponseType> {
+   /**
+    * 
+    * @param context the context of the api call
+    * @param response the response from the server
+    * @return the Response
+    * @throws IOException which occurs in case of I/O error
+    */
+   ResponseType apply(Context context, ResponseType response) throws IOException;
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/coreinterfaces/type/functional/ContextInitializer.java:21]: Is it expected to throw IOException while applying the function?

File: src/main/java/io/apimatic/coreinterfaces/type/functional/ContextInitializer.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package io.apimatic.coreinterfaces.type.functional;
+
+import java.io.IOException;
+import io.apimatic.coreinterfaces.http.Context;
+
+
+/**
+* Functional Interface to apply the context initialization function.
```
Comment (sufyankhanrao) [inline src/main/java/io/apimatic/coreinterfaces/type/functional/ContextInitializer.java:8]: update the description with the following,
Functional Interface to apply the context initialization function for the response models.

File: pom.xml [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
 	<groupId>io.apimatic</groupId>
 	<artifactId>core-interfaces</artifactId>
-	<version>0.1.3</version>
+	<version>0.1.4</version>
```
Comment (sufyankhanrao) [inline pom.xml:10]: No need to upgrade this version as `0.1.3` has not been released yet so this change will become a part of `0.1.3` now.

File: README.md [hunk 4]
```diff
@@ -42,6 +42,7 @@ To add a dependency on core interfaces using Maven, use the following:
 | [`Deserializer`](./src/main/java/io/apimatic/coreinterfaces/type/functional/Deserializer.java)                  | Functional interface to  apply the deserialization function           |
 | [`ExceptionCreator`](./src/main/java/io/apimatic/coreinterfaces/type/functional/ExceptionCreator.java)                  | Functional interface to  create the SDK exception             |
 | [`Serializer`](./src/main/java/io/apimatic/coreinterfaces/type/functional/Serializer.java)                  | Functional interface to  apply the serialization function              |
+| [`ContextInitializer`](./src/main/java/io/apimatic/coreinterfaces/type/functional/ContextInitializer.java)                  | Functional interface to  apply the context initialization function              |
```
Comment (sufyankhanrao) [inline README.md:45]: The same recommended description goes here too.

File: src/main/java/io/apimatic/coreinterfaces/type/functional/ContextInitializer.java (hunk 2 above)
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/coreinterfaces/type/functional/ContextInitializer.java]: Updated

File: pom.xml (hunk 3 above)
Comment (hamzashoukat94) [reply pom.xml]: done

File: README.md (hunk 4 above)
Comment (hamzashoukat94) [reply README.md]: done

File: src/main/java/io/apimatic/coreinterfaces/type/functional/ContextInitializer.java (hunk 1 above)
Comment (hamzashoukat94) [reply src/main/java/io/apimatic/coreinterfaces/type/functional/ContextInitializer.java]: Removed the throws clause

**Review by sufyankhanrao — APPROVED** [review]

**Pull request merged**